posthog/management/commands/recreate_sessions_table.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+
+import structlog
+from django.core.management.base import BaseCommand
+
+from posthog.clickhouse.client.connection import Workload
+from posthog.clickhouse.client.execute import sync_execute
+from datetime import datetime
+
+from posthog.models.sessions.sql import get_session_table_mv_select_sql
+
+logger = structlog.get_logger(__name__)
+
+
+SETTINGS = {
+    "max_execution_time": 3600  # 1 hour
+}
+
+
+@dataclass
+class RecreateSessionsTable:
+    month: datetime
+    use_offline_workload: bool
+
+    def execute(self, print_counts=False) -> None:
+        logger.info(f"Recreating sessions table for {self.month.strftime('%Y-%m')}")
+
+        events_where = f"toYYYYMM(timestamp) = {self.month.strftime('%Y%m')}"
+        sessions_where = f"toYYYYMM(min_timestamp) = {self.month.strftime('%Y%m')}"
+        select_query = get_session_table_mv_select_sql(source_column_mode="json_or_mat", extra_where=events_where)
+
+        if print_counts:
+            count_query = f"SELECT count(), uniq(session_id) FROM sharded_sessions WHERE {sessions_where}"
+            [(sessions_row_count, sessions_event_count)] = sync_execute(count_query, settings=SETTINGS)
+            logger.info(f"{sessions_row_count} rows and {sessions_event_count} unique session_ids in sessions table")
+
+        logging.info("Deleting the existing sessions for month %s", self.month.strftime("%Y-%m"))
+        sync_execute(
+            query=f"""ALTER TABLE sharded_sessions DROP PARTITION {self.month.strftime('%Y%m')}""",
+            workload=Workload.OFFLINE if self.use_offline_workload else Workload.DEFAULT,
+            settings=SETTINGS,
+        )
+
+        if print_counts:
+            count_query = f"SELECT count(), uniq(session_id) FROM ({select_query})"
+            [(sessions_row_count, sessions_event_count)] = sync_execute(count_query, settings=SETTINGS)
+            logger.info(f"{sessions_row_count} rows and {sessions_event_count} unique session_ids in events table")
+
+        logging.info("Writing the new sessions for day %s", self.month.strftime("%Y-%m"))
+        sync_execute(
+            query=f"""INSERT INTO writable_sessions {select_query}""",
```

Comment:
This might take a while, the backfill runs on 1 day at a time and takes ~20 minutes, so it's possible this will take 10 hours